File: mobile_app/lib/features/search/services/search_service.dart
import '../data/mock/mock_search_data.dart';
import '../data/models/search_result_model.dart';

class SearchService {
  Future<List<SearchResultModel>> searchOrganizations({
    String? bloodGroup,
    String? province,
    String? city,
  }) async {
    // Temporary delay.
    // This will later be replaced with the real API call.
    await Future.delayed(
      const Duration(milliseconds: 800),
    );

    List<SearchResultModel> results = List<SearchResultModel>.from(
      mockSearchResults,
    );

    // Blood Group Filter
    if (bloodGroup != null && bloodGroup.isNotEmpty) {
      results = results.where((item) {
        return item.bloodGroup.toLowerCase() ==
            bloodGroup.toLowerCase();
      }).toList();
    }

    // Province Filter
    if (province != null && province.isNotEmpty) {
      results = results.where((item) {
        return item.province.toLowerCase() ==
            province.toLowerCase();
      }).toList();
    }

    // City Filter
    if (city != null && city.isNotEmpty) {
      results = results.where((item) {
        return item.city.toLowerCase() ==
            city.toLowerCase();
      }).toList();
    }

    return results;
  }
}

File: lifelynk-web/backend/app/services/search_service.py
from typing import Any

from app.repositories.search_repository import SearchRepository


class SearchService:
    def __init__(self, repository: SearchRepository) -> None:
        self.repository = repository

    async def search_organizations(
        self,
        blood_group: str | None = None,
        province: str | None = None,
        city: str | None = None,
    ) -> list[dict[str, Any]]:
        return await self.repository.search_organizations(
            blood_group=blood_group,
            province=province,
            city=city,
        )